modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/transfer/ProcessExport.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.pc.core.transfer;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.pc.core.FolderZiper;
+import org.wso2.carbon.pc.core.ProcessCenterConstants;
+import org.wso2.carbon.pc.core.ProcessCenterException;
+import org.wso2.carbon.pc.core.ProcessStore;
+import org.wso2.carbon.pc.core.internal.ProcessCenterServerHolder;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProcessExport {
+    private List<String> exportedProcessList;
+    private static final Log log = LogFactory.getLog(ProcessExport.class);
+
+    /**
+     * @param processName
+     * @param processVersion
+     * @param exportWithAssociations
+     * @param user
+     * @return
+     * @throws Exception
+     */
+    public String initiateExportProcess(String processName, String processVersion, String exportWithAssociations,
+            String user) throws Exception {
+        exportedProcessList = new ArrayList<String>();
+        File exportsDir = new File(ProcessCenterConstants.PROCESS_EXPORT_DIR);
+        if (exportsDir != null) {
+            exportsDir.mkdirs();
+            Boolean exportWithAssociationsBool = Boolean.valueOf(exportWithAssociations);
+            try {
+                // save details about the exported zip and the core process
+                String exportRootPath =
+                        ProcessCenterConstants.PROCESS_EXPORT_DIR + "/" + processName + "-" + processVersion +
```

Comment:
Since you have used "/", does this work in windows?
Since this is in Java 7, can use Paths.